apps/prairielearn/elements/pl-dataframe/pl-dataframe.py

```diff
@@ -174,9 +174,24 @@ def render(element_html: str, data: pl.QuestionData) -> str:
     # Might be worth moving everything out of the CSS file and handle it all with the builtin styler.
     frame_style.set_table_attributes("class=pl-dataframe-table")
 
+    # Generate the initial HTML
+    frame_html_raw = frame_style.to_html()
+
+    # https://www.w3.org/WAI/WCAG21/Techniques/html/H63
+    html_root = lxml.html.fromstring(frame_html_raw)
+    for th in html_root.xpath("//th[contains(@class, 'blank')]"):
+        th.set("scope", "colgroup")
+    for th in html_root.xpath("//th[contains(@class, 'col_heading')]"):
+        th.set("scope", "col")
```

Comment:
Is this really necessary? I'd expect the nesting of `thead tr th` to be sufficient for communicating that a `<th>` is a column header, and the WCAG link you included above indicates the same:

> For simple tables that have the headers in the first row or column, it is sufficient to simply use the th elements without scope.

Reply:
https://github.com/PrairieLearn/PrairieLearn/pull/11842#discussion_r2058989724

Reply:
My comment was about `col`, not `colgroup`.

Even then: why is `colgroup` an appropriate scope to use here? That's generally used when a header spans multiple columns, like this this contrived example:

```html
<table>
  <thead>
    <tr>
      <th scope="colgroup" colspan="2">Personal Info</th>
      <th scope="col">Age</th>
    </tr>
    <tr>
      <th scope="col">First Name</th>
      <th scope="col">Last Name</th>
      <th scope="col">Age</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Alice</td>
      <td>Smith</td>
      <td>30</td>
    </tr>
  </tbody>
</table>
```

Based on your screenshot in https://github.com/PrairieLearn/PrairieLearn/pull/11842#discussion_r2058989724, that's not relevant here.

Reply:
According to the [example on the docs](https://www.w3.org/WAI/WCAG21/Techniques/html/H63)

```
  <tr>
    <td></td>
    <th scope="col">Name</th>
    <th scope="col">Phone Number</th>
    <th scope="col">City</th>
  </tr>
  ```
  
  When I read this last night, I misread the first element `<td></td>` as `<tr></tr>`. I'll fix this up now.

Reply:
Thanks! This same comment applies to the other places you added `scope="col"` in this PR. In general, if one is using `<thead><tr><th>...`, that shouldn't be necessary.

Reply:
The other place is interesting, as I had resolved the scope="col" errors first, and then then thead errors. I bet if I resolved them in the other order, that wouldn't have happened.

Reply:
Changed in 7390fae9c09d73fc00277f8bdc562dd71f642d55. This affects some first-party code too.